ssl-config/src/main/java/com/palantir/remoting/ssl/TrustStoreConfiguration.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting.ssl;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Optional;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Style.ImplementationVisibility;
+
+@JsonDeserialize(as = ImmutableTrustStoreConfiguration.class)
+@JsonSerialize(as = ImmutableTrustStoreConfiguration.class)
+@Value.Immutable
+@Value.Style(visibility = ImplementationVisibility.PACKAGE)
+public abstract class TrustStoreConfiguration {
```

Comment:
Note that, currently, this commit does not break existing config -- both `com.palantir.trust.TrustStores` and `com.palantir.trust.TrustStoreConfiguration` still exist but are deprecated and call into the new code, so existing configuration will work until we delete those deprecated files.

That being said, yes, I do think that this format seems correct/reasonable, and wouldn't anticipate any breaking changes to it (possibly additive in the future, but find that unlikely as well)